tc/sgx/trusted_worker_manager/enclave_wrapper/file_io_processor.cpp

```diff
@@ -0,0 +1,298 @@
+        result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+        return FAILED;
+    }
+
+    try {
+        ifstream fileStream (fileName);
+        if (!fileStream.is_open()) {
+            tcf::Log(TCF_LOG_ERROR, "Unable to open file %s\n", fileName.c_str());
+            resultStr = "FAILED TO OPEN: Unable to open file";
+            resultSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            fileStream.close();
+            return FAILED;
+        }
+        tcf::Log(TCF_LOG_INFO, "File %s OPENED successfully\n", fileName.c_str());
+        fileStream.close();
+
+    } catch (std::exception e) {
+        tcf::Log(TCF_LOG_ERROR, "Caught exception while opening file: %s\n", e.what());
+        resultStr = "FAILED TO OPEN: Caught Exception while opening file";
+        resultSize = resultStr.length();
+        result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+        return FAILED;
+    }
+
+    resultStr = "FILE OPEN SUCCESS";
+    resultSize = resultStr.length();
+    result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+    return SUCCESS;
+}
+
+// Closes file and updates status SUCCESS or FAILED in result parameter
+uint32_t FileClose(std::string fileName, uint8_t *result, size_t resultSize) {
+    string resultStr;
+    if (fileName.empty()) {
+        tcf::Log(TCF_LOG_ERROR, "Filename is empty\n");
+        resultStr = "FAILED TO CLOSE: Filename is empty";
+        resultSize = resultStr.length();
+        result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+        return FAILED;
+    }
+
+    try {
+        ifstream fileStream (fileName);
+        if (!fileStream.is_open()) {
+            tcf::Log(TCF_LOG_ERROR, "Unable to close file %s\n", fileName.c_str());
+            resultStr = "FAILED TO CLOSE: Unable to close file";
+            resultSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            return FAILED;
+        }
+
+        fileStream.close();
+        tcf::Log(TCF_LOG_INFO, "File %s CLOSED successfully\n", fileName.c_str());
+
+    } catch (std::ifstream::failure e) {
+            tcf::Log(TCF_LOG_ERROR, "Caught exception while Closing file: %s\n", e.what());
+            resultStr = "FAILED TO CLOSE: Caught exception while closing file";
+            resultSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            return FAILED;
+    }
+
+    resultStr = "FILE CLOSE SUCCESS";
+    resultSize = resultStr.length();
+    result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+    return SUCCESS;
+}
+
+// Reads file, store content of file in buffer and updates status
+// SUCCESS or FAILED in result parameter
+uint32_t FileRead(std::string fileName, uint8_t *result, size_t resultSize,
```

Comment:
What is the purpose of the `resultSize` parameter? It is unused. The incoming parameter value is always overwritten in the function and the incoming value is never used.
You should test that the string copied to `result` never exceeds the input value of `resultSize` (including the `NULL` character).
Also document in in the function comments that `resultSize` is the maximum size of the `result` buffer.

Reply:
I agree, we need to use resultSize as max threshold value of the string value that can be copied to result parameter. 
I will add this change along with documentation.

Reply:
Keep `resultSize` for FileRead().
Remove `resultSize` for FileWrite()  (above) as the buffer is not modified.

Reply:
For `FileRead()` also document what the return value is. Is it 0/non-zero for success/failure? Or is the number of bytes actually read?

Reply:
> For `FileRead()` also document what the return value is. Is it 0/non-zero for success/failure? Or is the number of bytes actually read?

return value is string with value either SUCCESS or FAILURE : REASON FOR FAILURE.